lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -1540,12 +1607,398 @@ void amf_app_handle_initial_context_setup_rsp(
       }
     }
   }
 
   // update UE state
   ue_context->mm_state = REGISTERED_CONNECTED;
 }
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    amf_app_pdu_session_modification_request()                    **
+ **                                                                        **
+ ** Description: Send the PDU modification request to gnodeb               **
+ **                                                                        **
+ ** Inputs:  pdu_sess_mod_req:   pdusession modification request           **
+ **      ue_id:      ue identity                                           **
+ **                                                                        **
+ **      Return:    RETURNok, RETURNerror                                  **
+ **                                                                        **
+ ***************************************************************************/
+int amf_app_pdu_session_modification_request(
+    itti_n11_create_pdu_session_response_t* pdu_sess_mod_req,
+    amf_ue_ngap_id_t ue_id) {
+  nas5g_error_code_t rc = M5G_AS_SUCCESS;
+
+  amf_nas_message_t msg = {};
+  uint32_t bytes = 0;
+  uint32_t len = 0;
+  bstring buffer;
+  imsi64_t imsi64 = INVALID_IMSI64;
+  amf_context_t* amf_ctxt_p = NULL;
+  ue_m5gmm_context_s* ue_context = NULL;
+  std::shared_ptr<smf_context_t> smf_ctx;
+  uint32_t buf_len = 0;
+
+  ue_context = amf_ue_context_exists_amf_ue_ngap_id(ue_id);
+
+  if (!ue_context) {
+    OAILOG_ERROR(LOG_AMF_APP,
+                 "ue context not found for the ue_id:" AMF_UE_NGAP_ID_FMT,
+                 ue_id);
+    return M5G_AS_FAILURE;
+  }
+
+  smf_ctx = amf_get_smf_context_by_pdu_session_id(
+      ue_context, pdu_sess_mod_req->pdu_session_id);
+  if (!smf_ctx) {
+    OAILOG_ERROR(LOG_AMF_APP,
+                 "Smf context is not exist UE ID:" AMF_UE_NGAP_ID_FMT, ue_id);
+    return M5G_AS_FAILURE;
+  }
+
+  // Message construction for PDU session modification
+  msg.security_protected.plain.amf.header.extended_protocol_discriminator =
+      M5G_MOBILITY_MANAGEMENT_MESSAGES;
+  msg.security_protected.plain.amf.header.message_type =
+      M5GMessageType::PDU_SESSION_MODIFICATION_COMMAND;
+  msg.header.security_header_type =
+      SECURITY_HEADER_TYPE_INTEGRITY_PROTECTED_CYPHERED;
+  msg.header.extended_protocol_discriminator = M5G_MOBILITY_MANAGEMENT_MESSAGES;
+  msg.header.sequence_number =
+      ue_context->amf_context._security.dl_count.seq_num;
+  msg.security_protected.plain.amf.msg.pdu_sess_mod_cmd
+      .extended_protocol_discriminator.extended_proto_discriminator =
+      M5G_SESSION_MANAGEMENT_MESSAGES;
+  msg.security_protected.plain.amf.msg.pdu_sess_mod_cmd.pdu_session_identity
+      .pdu_session_id = pdu_sess_mod_req->pdu_session_id;
+  msg.security_protected.plain.amf.msg.pdu_sess_mod_cmd.pti.pti =
+      smf_ctx->smf_proc_data.pti;
```

Comment:
AMF is initiating "Pdu session modification". So where is PTI generated? How is this maintained for requests received from UE and while sending initial messages from AMF